server/src/main/java/com/linecorp/centraldogma/server/management/ServerStatus.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.management;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+/**
+ * The status of the server.
+ */
+@JsonInclude(Include.NON_NULL)
+public final class ServerStatus {
+    private final boolean writable;
+    private final boolean replicating;
+
+    // TODO(trustin): Add more properties, e.g. method, host name, isLeader and config.
+
+    /**
+     * Creates a new instance with the specified properties.
+     */
+    @JsonCreator
+    public ServerStatus(@JsonProperty("writable") @Nullable Boolean writable,
+                        @JsonProperty("replicating") @Nullable Boolean replicating) {
+        writable = firstNonNull(writable, true);
+        replicating = firstNonNull(replicating, true);
+        if (writable && !replicating) {
```

Comment:
I often find it confusing to determine which combination of writable and replicable is possible. What do you think about adding an enum instead?
This would eliminate confusion for both us and users, ensuring that incorrect combinations are not input.

```java
enum ServerStatus {
  READABLE(false, false),
  REPLICABLE(false, true),
  WRITABLE(true, true);
}
```

Reply:
Good idea. I prefer renaming some of them for clarity. How about?
```java
enum ServerStatus {
  READONLY(false, false),
  REPLICATION_ONLY(false, true),
  WRITABLE(true, true);
}
```

Reply:
That looks perfect. 💯 💯 💯